composer/benchmarker/benchmarker.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+from typing import List, Optional, Sequence
+
+import torch
+import torch.distributed
+
+from composer.benchmarker.benchmarker_hparams import BenchmarkerHparams
+from composer.callbacks.benchmarker import Benchmarker as BenchmarkerCallback
+from composer.core.logging.base_backend import BaseLoggerBackend
+from composer.core.precision import Precision
+from composer.datasets.hparams import DataloaderSpec
+from composer.datasets.synthetic import SyntheticDataLabelType, SyntheticDataset
+from composer.models.base import BaseMosaicModel
+from composer.trainer.devices import DeviceCPU, DeviceGPU
+from composer.trainer.trainer import Trainer
+
+NUM_PROFILING_STEPS = 250  # needs to be high enough to be able to properly measure time
+
+
+class Benchmarker:
```

Comment:
For ease of usability, would it be possible to have the benchmarker take the same yaml (i.e. hparams) file as is used for training, plus a few additional benchmarker-specific arguments (e.g. num samples, and for the time being until #103 is done, synthetic dataset params). I'm thinking it would be really neat if you could invoke it like `benchmarker --num samples 100 -f path/to/training_hparams.yaml`

Reply:
In the future there is a chance that this benchmarked will just merge with the trainer but for simplicity I made it a separate class for now